### PR TITLE
feat(tui): search and order the worktree picker by recent work

### PR DESCRIPTION
**Pull request by artyomsv**
## Summary

The setup dialog's **Worktree** field listed worktrees in git's own order — alphabetical by admin directory — and drew each row as `<branch>  <path>` cut at the right edge. On a 58-worktree monorepo that put the checkout being worked on that day at **row 49 of a six-row window**, under a branch name sharing no word with its folder, with the folder name itself truncated off the row's tail. The row was on screen and unfindable; it read as a worktree the picker was hiding.

Three changes, one per cause:

- **Type-to-search** on the focused field. Matches branch **or path**, case-insensitively — the folder is often the name a worktree is known by, and it can share no word with the branch. `Esc` clears the search before it closes the dialog, `Enter` commits the match and restores the full list with the choice marked. `j`/`k` become letters (the trade the new-branch name field already makes); `Up`/`Down` still move.
- **Order by recent work.** The worktree the pane being split sits in comes first, tagged `(current)` — a second agent beside the first on one checkout is the usual reason to open this field. The rest sort by their branch's last commit, newest first.
- **Cut the path at its head, not its tail.** Every worktree of one repository shares a parent, so the head carries no information and the tail is the folder name the user recognises.

### Notes for review

- The commit date rides the **existing** `for-each-ref` call that already lists branch names (one tab-separated field — no extra subprocess, no per-branch `git log`). The daemon joins it onto `WorktreeInfo.CommitTime` by branch name, because the join key is git's spelling on both sides only on the machine that ran git; the wire's `Branches` stays a list of names for the collision check and for older clients.
- A daemon too old to send dates yields all zeroes, and `sort.SliceStable` leaves git's order intact — the picker degrades to today's behaviour rather than shuffling.
- `worktreePathInside` folds case and unifies separators on **both** sides: the pane's CWD is whatever OSC 7 reported and the path is what git printed, and on Windows those differ in both. Containment is tested on a separator boundary, so a pane in `feat-ab` is not read as inside `feat-a`.
- The search rides the field's **label** row, so it costs no list row and `worktreeVisibleRows`' budget is untouched. It is sanitized at render like every other value on these rows — it is the value's one draw site, and a C1 introducer or a bidi override is printable.
- The key handler's inert-guard moved from "no rows" to `worktreeFieldInteractive()`: a search matching nothing also has no rows, and the way out of it is a backspace the handler must still accept.

## Test Plan

- `go vet ./...`, `go build ./cmd/quil ./cmd/quild` — clean.
- `go test -race -timeout 300s ./...` and `go test -tags=integration -race ./...` — the exact CI commands, both green, run again after rebasing onto `master` (the sandbox feature landed in four of the same files).
- 19 new tests in `internal/tui/worktree_find_test.go` covering the ordering (including the all-zero-dates fallback and the separator-boundary case), the search (typing, folder match, case, `j`/`k`, backspace, the two-stage `Esc`, commit-and-restore, the empty match, the ingest bound), both resets, and the four render states.
- `TestBranches_RealGit_CarriesEachBranchesCommitTime` pins the format string against **real git** with a fixed `GIT_COMMITTER_DATE` — a stub agrees with any format, so nothing else could catch a silently-zero date.
- Smoke-tested in a dev build against the real 58-worktree monorepo: the list now opens on branches committed to today, and typing `verific` narrows it to the verification worktrees immediately.

### CODE REVIEW

File: internal/tui/dialog.go [hunk 1]
```diff
@@ -4834,8 +4858,51 @@ func (m Model) handleSetupWorktreeKey(p *plugin.PanePlugin, key string) (tea.Mod
...
+		// Bounded at ingest: the value is rendered on every frame, and the
+		// row it renders on truncates rather than wraps, so an unbounded one
+		// would be invisible past the edge and costly to draw regardless.
+		if utf8.RuneCountInString(key) == 1 && utf8.RuneCountInString(m.worktreeFilter) < worktreeFilterCap {
+			m.worktreeFilter += key
+			// Every change starts the cursor over on the first match: the row
+			// it was on may no longer be in the list, and the first match is
+			// the one Enter is most often meant for.
+			m.worktreeCursor, m.worktreeScroll = 0, 0
+		}
```
[the remark this answers is not in the document]
Comment (artyomsv) [reply internal/tui/dialog.go:4914]: Confirmed and fixed in 5301ceb.

Verified the claim before acting on it rather than taking the suggestion as given, and it holds: `handleConfirmKey` (dialog.go) carries the comment "Bubble Tea v2 reports the space key as `space` while a pasted or synthesised one can arrive as ` `", and `handleRenameKey` (model.go) does exactly this translation for the tab-rename input. So a real press was dropped and only the pasted spelling worked.

Applied as the same two-line normalisation those two use, with three regression tests: the `"space"` name, the literal `" "` (which already worked, pinned so a rewrite cannot drop one spelling while keeping the other), and a search that has to *match* a path containing a space, since accepting the key is only useful if the match then lands. All three were watched failing first — the name spelling gave `"ab"` for `a`,`space`,`b`.

Left `handleWorktreeNameKey` alone deliberately: it has the same gap, but it names a git branch, and git refuses a branch name containing a space, so there is nothing to reach there.
